documentation/index.md

```diff
@@ -2,41 +2,79 @@
 
 ## Events
 
-#### EVENT_CHANNEL_UPDATED
+### EVENT_CHANNEL_UPDATED
 
-Listens for any channel updates. Event contains the following:
- - channelId: The channel ID of the app instance.
- - deviceToken: (deprecated, iOS only) The device token.
- - pushToken: The push token.
+Event sent when the channel is updated or created.
+ - channelId: String. The channel ID of the app instance.
+ - deviceToken: (deprecated, iOS only) String. The device token.
+ - pushToken: String. The push token.
 
 ```
     Airship.addEventListener(Airship.EVENT_CHANNEL_UPDATED, function(e) {
-        Ti.API.info('Channel Updated: ' + Airship.channelId)
-    });
+        Ti.API.info('Channel Updated: ' + e.channelId)
+    })
 ```
 
-#### EVENT_PUSH_RECEIVED
+### EVENT_NOTIFICATION_OPT_IN_CHANGED
 
-Listens for any push received events. Event contains the following:
- - message: The push alert message.
- - extras: Map of all the push extras.
- - notificationId: (Android only) The ID of the posted notification.
+Event sent when the notification opt-in status changes.
+- optIn: Boolean. True if the notifications are opted-in, otherwise false.
+- authorizedSettings: Object. iOS Only.
+    - alert: Boolean. True if alert is allowed, otherwise false.
+    - badge: Boolean. True if badge is allowed, otherwise false.
+    - sound: Boolean. True if sound is allowed, otherwise false.
+    - criticalAlert: Boolean. True if critical alerts are allowed, otherwise false.
+    - announcement: Boolean. True if announcements is allowed, otherwise false.
+    - lockScreen: Boolean. True if lock screen notifications are allowed, otherwise false.
+    - notificationCenter: Boolean. True if notification center notifications are allowed, otherwise false.
+    - carPlay: Boolean. True if car play notifications are allowed, otherwise false.
+
+```
+    Airship.addEventListener(Airship.EVENT_NOTIFICATION_OPT_IN_CHANGED, function(e) {
+        Ti.API.info('OptIn: ' + e.optIn)
+    })
+```
+
+### EVENT_PUSH_RECEIVED
+
+Event sent when a push is received.
+ - message: String. The message's alert.
+ - title: String. The message's title.
+ - extras: Object. The message's extras.
+ - notificationId: String. The notification ID.
+ - receivedInForeground: Boolean. True if received in foreground, otherwise false.
```

Comment:
Are we using this elsewhere? If not I wonder if we might want to call it `isForeground`, to match other events/platforms.

Reply:
So the other response event has `isForeground` since it defines if it will foreground the app. This just indicates in what state it was received in. I am bad at naming so I will take suggestions.